homeassistant/components/yale_smart_alarm/const.py

```diff
@@ -14,8 +14,10 @@
 )
 
 CONF_AREA_ID = "area_id"
+CONF_LOCK_CODE_DIGITS = "lock_code_digits"
```

Comment:
Can you help me understand why the # of digits is needed vs just getting the # of digits from the actual code entered? I assume there is a good reason, i just can't tell from this PR.

Reply:
In the UI for locks you can set it as a setting as a small check when someone is typing so sure, not something very important but as it's an option to use I just set it in for use when I get locks in (the user can choose not to set his lockcode in the config).

Reply:
I'm not sure I understand, can you elaborate?

I get the idea of asking the user a lock code with a fixed # of digits in the lock UI.

I'm just not sure why it's needed as an option int he config flow if it always matches the lock code entered. What would happen if it wasn't here, and wasn't store since it can be inferred with len(lock code). Is it here as a protection in case the user enters the wrong length code?

Reply:
First off it's not an absolute requirement and therefore it's part of options.
It's to feed the "code_format" option of lock platform https://github.com/home-assistant/core/blob/dev/homeassistant/components/lock/__init__.py#L98 primarily but is also used to check the code supplied in config_flow that it's the right length.
As it's available to use as a control then my personal opinion is to use it as such.

Reply:
To clarify: I get the reason why you need a code format in general, and why it is useful to customize it -- totally makes sense. I was asking more why it needs to exist here since its tied to the other code entered.  I think its because the code is also entered here at the same time it was a bit confusing.   (e.g. As a user, if you go into the options and change `CONF_LOCK_CODE_DIGITS` from 4 to 6 I also have to change `CONF_CODE` to a 6 digit code.   If I change the length of `CONF_CODE` from a 4 digit code to a 6 digit code, then I also have to change `CONF_LOCK_CODE_DIGITS` from 4 to 6)

The answer is that you will use this in the config flow in follow up PRs, makes sense. Thank you for clarifying.

Do you need additional validation on the code besides just checking it's length? or is this something planned to do in the future verifying with the lock platform itself?

Reply:
The rest will be done in the platform. As the length can be different it needs to be asked in the flow itself from the user input.